app/src/main/java/com/gathering/android/event/home/EventListViewModel.kt

```diff
@@ -66,9 +79,32 @@ class EventListViewModel @Inject constructor() : ViewModel() {
 
             }
         }
+
+        if (sortType == SortType.SORT_BY_DATE) {
+            filteredEventList = filteredEventList.sortedBy { it.date }
+        } else {
+            filteredEventList = filteredEventList.sortedWith(object : kotlin.Comparator<Event> {
```

Comment:
take this SAM to a class that implements `Comperator`. it can be called `EventLocationComperator`